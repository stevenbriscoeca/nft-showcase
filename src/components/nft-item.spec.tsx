import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import NftItem from "./ntf-item";

describe("nft-item", () => {
  it(" should have correct name when setting the name prop", async () => {
    // ARRANGE
    render(<NftItem name="lol"></NftItem>);

    // ACT
    const element = screen.getByTestId("name");

    // ASSERT
    expect(element).toHaveTextContent("lol");
  });

  it("should have correct image when setting the image prop", async () => {
    // ARRANGE
    render(<NftItem image="lmao.gif"></NftItem>);

    // ACT
    const imageSrc = screen.getByTestId("image").getAttribute("src");

    // ASSERT
    expect(imageSrc).toEqual("lmao.gif");
  });

  it("should have correct url when setting the url prop", async () => {
    // ARRANGE
    render(<NftItem url="https://opensea.io/"></NftItem>);

    // ACT
    const imageSrc = screen.getByTestId("url").getAttribute("href");

    // ASSERT
    expect(imageSrc).toEqual("https://opensea.io/");
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <NftItem name="lol" url="https://opensea.io/" image="lmao.gif"></NftItem>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
