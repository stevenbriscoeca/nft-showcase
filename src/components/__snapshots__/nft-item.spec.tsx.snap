// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nft-item matches snapshot 1`] = `
<DocumentFragment>
  <div
    class=" bg-amber-400 rounded"
  >
    <a
      class="font-bold underline text-white"
      data-testid="url"
      href="https://opensea.io/"
    >
      <span
        data-testid="name"
      >
        lol
      </span>
      <img
        alt=""
        data-testid="image"
        src="lmao.gif"
      />
    </a>
  </div>
</DocumentFragment>
`;
